zkstats/ops.py

```diff
@@ -25,24 +26,35 @@ def ezkl(self, x: list[torch.Tensor]) -> IsResultPrecise:
 
 class Where(Operation):
     @classmethod
-    def create(cls, x: list[torch.Tensor], error: float) -> 'Where':
+    def create(cls, x: list[torch.Tensor], error: float, precal_witness:dict = None, op_dict:dict = {}) -> 'Where':
         # here error is trivial, but here to conform to other functions
-        return cls(torch.where(x[0],x[1], MagicNumber ),error)
+        # just dummy result, since not using it anyway because we dont want to expose direct result from where
+        return cls(torch.tensor(1),error)
     def ezkl(self, x:list[torch.Tensor]) -> IsResultPrecise:
-        bool_array = torch.logical_or(x[1]==self.result, torch.logical_and(torch.logical_not(x[0]), self.result==MagicNumber))
-        # print('sellll: ', self.result)
-        return torch.sum(bool_array.float())==x[1].size()[1]
+        return torch.tensor(True)
+
 
 
 class Mean(Operation):
     @classmethod
-    def create(cls, x: list[torch.Tensor], error: float) -> 'Mean':
+    def create(cls, x: list[torch.Tensor], error: float, precal_witness:dict = None, op_dict:dict = {} ) -> 'Mean':
         # support where statement, hopefully we can use 'nan' once onnx.isnan() is supported
-        return cls(torch.mean(x[0][x[0]!=MagicNumber]), error)
+        if precal_witness is None:
+            # this is prover
+            # print('provvv')
+            return cls(torch.mean(x[0][x[0]!=MagicNumber]), error)
+        else:
+            # this is verifier
+            # print('verrrr')
+            if 'Mean' not in op_dict:
+                return cls(torch.tensor(precal_witness['Mean_0'][0]), error)
+            else:
+                return cls(torch.tensor(precal_witness['Mean_'+str(op_dict['Mean'])][0]), error)
+
 
     def ezkl(self, x: list[torch.Tensor]) -> IsResultPrecise:
         x = x[0]
-        size = torch.sum((x!=MagicNumber).float())
+        size = torch.sum(torch.where(x!=MagicNumber, 1.0, 0.0))
```

Comment:
why are we using `torch.where` here over `x != y`?

Reply:
not really different...